File: backend/src/InsuraTech.Application/Claims/Commands/RegisterClaim/RegisterClaimCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InsuraTech.Application.Claims.DTOs;
using InsuraTech.Domain.Claims;
using MediatR;

namespace InsuraTech.Application.Claims.Commands.RegisterClaim
{
    public sealed record RegisterClaimCommand : IRequest<ClaimResponse>
    {
        public Guid    PolicyId            { get; init; }
        public ClaimType Type              { get; init; }
        public decimal ClaimedAmount       { get; init; }
        public DateOnly IncidentDate       { get; init; }
        public string  Description         { get; init; } = null!;
        public string  ResponsibleUser     { get; init; } = null!;

        /// <summary>Se inyecta desde el controller vía JWT. No viene en el body del request.</summary>
        public Guid?   CreatedByAdvisorId  { get; init; }
    }
}

File: backend_v2/src/main/java/com/insuratech/application/claims/commands/RegisterClaimCommand.java
package com.insuratech.application.claims.commands;

import com.insuratech.domain.claims.ClaimType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import java.math.BigDecimal;
import java.time.LocalDate;

public record RegisterClaimCommand(
    @NotBlank String policyId,
    @NotNull ClaimType type,
    @NotNull @Positive BigDecimal claimedAmount,
    @NotNull LocalDate incidentDate,
    @NotBlank String description,
    @NotBlank String responsibleUser,
    String createdByAdvisorId
) {}
